mavros_extras/src/plugins/debug_value.cpp

```diff
@@ -0,0 +1,252 @@
+		value_msg->value_float = value.value;
+
+		named_value_float_pub.publish(value_msg);
+	}
+
+	/**
+	 * @brief Handle NAMED_VALUE_INT message.
+	 * Message specification: http://mavlink.org/messages/common/#NAMED_VALUE_INT
+	 * @param msg	Received Mavlink msg
+	 * @param value	NAMED_VALUE_INT msg
+	 */
+	void handle_named_value_int(const mavlink::mavlink_message_t *msg, mavlink::common::msg::NAMED_VALUE_INT &value)
+	{
+		auto value_msg = boost::make_shared<mavros_msgs::DebugValue>();
+
+		std::string name = mavlink::to_string(value.name);
+		ros::Time stamp = m_uas->synchronise_stamp(value.time_boot_ms);
+
+		ROS_DEBUG_STREAM_NAMED("NAMED_VALUE_INT", "timestamp: " << stamp
+									<< " name: " << name
+									<< " value: " << value.value);
+
+		value_msg->header.stamp = stamp;
+		value_msg->name = name;
+		value_msg->index = -1;
+		value_msg->type = value_msg->TYPE_NAMED_VALUE_INT;
+		value_msg->value_float = value.value;
+
+		named_value_float_pub.publish(value_msg);
+	}
+
+	/* -*- callbacks -*- */
+
+	/**
+	 * @brief Debug callbacks
+	 * @param req	pointer to mavros_msgs/Debug.msg being published
+	 */
+	void debug_cb(const mavros_msgs::DebugValue::ConstPtr &req)
+	{
+		switch (req->type) {
+		case mavros_msgs::DebugValue::TYPE_DEBUG:
+		{
+			mavlink::common::msg::DEBUG debug {};
+
+			debug.time_boot_ms = req->header.stamp.toNSec() / 1000000;
+			debug.ind = req->index;
+			debug.value = req->value_float;
+
+			UAS_FCU(m_uas)->send_message_ignore_drop(debug);
+			break;
+		}
+		case mavros_msgs::DebugValue::TYPE_DEBUG_VECT:
+		{
+			mavlink::common::msg::DEBUG_VECT debug {};
+
+			debug.time_usec = req->header.stamp.toNSec() / 1000;
+			std::copy(req->name.begin(), req->name.end(), debug.name.data());
+
+			// [[[cog:
+			// for a,b in zip("xyz", range(0,3)):
+			//     cog.outl("debug.%s = req->data[%s];" % (a, b))
+			// ]]]
+			debug.x = req->data[0];
+			debug.y = req->data[1];
+			debug.z = req->data[2];
+			// [[[end]]] (checksum: f4918ce98ca3183f93f6aff20d4ab7ec)
+
+			UAS_FCU(m_uas)->send_message_ignore_drop(debug);
+			break;
+		}
+		case mavros_msgs::DebugValue::TYPE_DEBUG_ARRAY:
+		{
+			// @todo
+			return;
+		}
+		case mavros_msgs::DebugValue::TYPE_NAMED_VALUE_FLOAT:
+		{
+			mavlink::common::msg::NAMED_VALUE_FLOAT value {};
+
+			value.time_boot_ms = req->header.stamp.toNSec() / 1000000;
+			std::copy(req->name.begin(), req->name.end(), value.name.data());
+			value.value = req->value_float;
+
+			UAS_FCU(m_uas)->send_message_ignore_drop(value);
+			break;
+		}
+		case mavros_msgs::DebugValue::TYPE_NAMED_VALUE_INT:
+		{
+			mavlink::common::msg::NAMED_VALUE_INT value {};
+
+			value.time_boot_ms = req->header.stamp.toNSec() / 1000000;
+			std::copy(req->name.begin(), req->name.end(), value.name.data());
+			value.value = req->value_int;
+
+			UAS_FCU(m_uas)->send_message_ignore_drop(value);
+			break;
+		}
+		default:
+			ROS_ERROR_NAMED("debug", "Wrong debug type (%d). Droping!...", req->type);
+			return;
+		}
+	}
```

Comment:
You may move common send msg out of switch. mavlink::Message *ptr should be fine.